### ISSUE TITLE
Imports with Encrypted Drives in the %PATH%

### ISSUE THREAD

**Issue by twangboy** [pywin32/pypiwin32#63]
One of our community members recently reported an issue (https://github.com/saltstack/salt/issues/41298) where the `salt-minion` service will not start when there is a path to an encrypted drive (BitLocker) in the %PATH% environment variable. We've narrowed this down to an issue with PyWin32.

Here's how we're setting up the environment. Following his instructions:
1. Encrypt a Drive (Not C:)
   I created a 1GB disk drive and added it to my VM, then Enabled BitLocker and encrypted the Drive. The drive is empty.
2. Edit the %PATH% to point to the encrypted drive. The order doesn't matter.
3. Lock the drive (`manage-bde -lock e:` if `e:` is your encrypted drive)

At this point, open Python and import any Win32 library:
```
>>> import win32file
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: DLL load failed: This drive is locked by BitLocker Drive Encryption. You must unlock this drive from Control Panel.
```

If I unlock the drive, the import works fine. If I remove `E:\` from the path, it works fine.

I believe this is occurring because this specific BitLocker Error wasn't taken into account when handling possible returns while trying to find the appropriate DLLs.

PyWin32 seems to be the only module affected by the BitLocker drive being in the Path.

Versions:
Windows: 10
Python: 2.7.13
pypiwin32: 219

**Comment by xoviat** [comment]
I strongly suspect this is caused by the custom loader used by pywin32.

**Comment by twangboy** [comment]
@xoviat What is the way forward on this issue?

**Comment by xoviat** [comment]
The way forward on this issue is to wait until salt is upgraded to python3 so that I don't spend time fixing issues that may be resolved anyway.

**Comment by xoviat** [comment]
@wecsam Appears to be experiencing this issue as well. David, can you reproduce this issue using win32core/win32compat?

```
pip uninstall -y pypiwin32 pywin32
pip install git+https://github.com/pywin32/win32core.git win32compat
```

Note that you will need to have MS build tools 2015/2017 for Python 3 (actually 2017 is a bit flaky because it's in beta).

**Comment by wecsam** [comment]
@xoviat I was not having trouble with BitLocked drives, but I'm happy to oblige. I'm running Python 3.6.2 32-bit. For now, while running the second command, I am getting:

    error: could not create 'c:\program files (x86)\python36-32\Lib\site-packages\win32': Access is denied

I will investigate that later.

**Comment by xoviat** [comment]
Are you running `pip` as an administrator? What is the traceback?

**Comment by xoviat** [comment]
The user option is, if you don't trust me, you can use the `--user` flag.

`pip install --user git+https://github.com/pywin32/win32core.git win32compat`

**Comment by wecsam** [comment]
Nope, it turns out that the file permissions were messed up on my computer. I fixed it and was able to install `win32compat`. I then launched interactive Python with a BitLocked drive in PATH, entered `import win32file`, and pressed Enter. There was no error.

**Comment by xoviat** [comment]
FYI there are two packages here. The pip command installs the 'win32core'
package directly from the source repository and the 'win32compat' package
from pypi. Please report any issues you have with those packages.

2017-08-24 15:01 GMT-05:00 David Tsai <notifications@github.com>:
